### DISCUSSION TITLE
Vs code setup error

### DISCUSSION THREAD

**Discussion by ArnaudBand** [Cyfrin/foundry-full-course-cu#1731] [Q&A] [answered]
Hello everyone, 
I'm learning this course using Vscode as Ide but I have an issue that I failed to fix. This is the error: `Expected token Semicolon got 'LParen'solc`.

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/89846108/4ea0f78c-e110-4a77-bf91-cbe213e10984)
It always appears in the first parentheses on the smart contract. Anyone who knows how to fix it. I appreciate.

**Comment by EngrPips** [comment]
Please show your full-screen @ArnaudBand.

**Reply by ArnaudBand** [reply]
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/89846108/2e90b36e-f0fe-433b-939b-492ab1f557e3)
This is the entire screen of Vscode, It does not appear only on this project.

**Reply by EngrPips** [reply]
Please take a screenshot when the `Expected token semicolon got 'LParen'` isn't showing on the screen so I can see the whole contract.

**Comment by urosognjenovic** [comment]
What compiler version are you using? Make sure you are using the latest compiler version.

**Reply by EngrPips** [reply]
Could that be the issue?

**Reply by urosognjenovic** [reply]
Yes, it could. You can try it with this contract:

```
// SDPX-License-Identifier: MIT
pragma solidity 0.4.24;

contract Temp {
    error MyError();
}
```
Solidity defaults to showing the error on the first parenthesis.

In this case, custom errors are not supported by Solidity 0.4.24.

So the solution is to use the latest version of Solidity compiler or to use a caret (for example ^0.8.19).

**Reply by EngrPips** [reply]
Oh I see, Thank you for the insight.

**Reply by ArnaudBand** [reply]
Sorry for the delayed response; I was participating in the Chainlink hackathon. I always use the latest version of Solidity.

```solidity
//SPDX-License-Identifier: MIT
pragma solidity ^0.8.19;
```

Which extensions do you use when you're running solidity?
@EngrPips @thebraveman17

**Reply by EngrPips** [reply]
Hello @ArnaudBand, I primarily use `Solidity Visual Developer` and one other one by `Nomic Foundation`

<img width="1440" alt="Screenshot 2024-06-06 at 15 34 48" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95760918/2b97ff29-e908-47bf-8b99-1a7f3cdb5e58">

**Reply by ArnaudBand** [reply, marked as answer]
I appreciate everyone who has helped me. Everything is running smoothly now with no error messages.

**Reply by EngrPips** [reply]
You are welcome champ. Keep crushing it!
